### DISCUSSION TITLE
Cannot import AutoML models from mlforecast.auto

### DISCUSSION THREAD

**Discussion by egolub22** [Nixtla/mlforecast#362] [Q&A] [answered]
I am trying to create a solution that utilizes the library of "auto" models in mlforecast (https://nixtlaverse.nixtla.io/mlforecast/auto.html), in order to streamline hyper-parameter tuning. However, when I go to import the models in the way that is shown in Nixtla's tutorials  (https://nixtlaverse.nixtla.io/mlforecast/docs/how-to-guides/hyperparameter_optimization.html) it indicates that the mlforecast.auto module does not exist. Indeed, when I checked what modules were in mlforecast, auto was not among them, and none of the existing modules contained any of the auto models.

`
Import statements:

    import os
    import numpy as np
    import pandas as pd
    import optuna
    import lightgbm as lgb
    import warnings
    from importlib.metadata import version
    from utilsforecast.evaluation import evaluate
    from statsforecast import StatsForecast
    from statsforecast.models import (
        AutoARIMA,
        AutoETS,
        AutoCES,
        AutoTheta,
        Theta,
        OptimizedTheta,
        DynamicTheta,
        DynamicOptimizedTheta,
        MSTL,
        SeasonalExponentialSmoothingOptimized)
    
    from mlforecast import MLForecast
    from mlforecast.target_transforms import Differences
    from mlforecast.auto import (
        AutoLightGBM,
        AutoMLForecast,
        AutoModel,
        AutoRidge,
        ridge_space,
    )
    
    import plotly
    from plotly.subplots import make_subplots
    import plotly.graph_objs as go
    from utilsforecast.losses import *
    
    version('plotly')
`

Error:
<img width="905" alt="Import_error" src="https://github.com/Nixtla/statsforecast/assets/43010144/3b1b62cf-26c8-46e1-902d-6c4073abf32e">

`
Checking for packages:

    import pkgutil
    
    # this is the package we are inspecting -- for example 'email' from stdlib
    import mlforecast
    
    package = mlforecast
    for importer, modname, ispkg in pkgutil.iter_modules(package.__path__):
        print("Found submodule %s (is a package: %s)" % (modname, ispkg))
`

Output:
<img width="353" alt="package_output" src="https://github.com/Nixtla/statsforecast/assets/43010144/b91c2e08-b5e6-433c-9475-0f153550e6f5">

**Comment by jmoralez** [comment, marked as answer]
The auto module was added in [0.12.1](https://github.com/Nixtla/mlforecast/releases/tag/v0.12.1), which version are you using?

**Reply by egolub22** [reply]
I did not specify the version type when installing and it defaulted to an earlier version type for some reason. Thanks again for the help
